crates/oxc_language_server/src/capabilities.rs

```diff
@@ -0,0 +1,137 @@
+use tower_lsp::lsp_types::{
+    ClientCapabilities, CodeActionKind, CodeActionOptions, CodeActionProviderCapability, OneOf,
+    ServerCapabilities, TextDocumentSyncCapability, TextDocumentSyncKind, WorkDoneProgressOptions,
+    WorkspaceFoldersServerCapabilities, WorkspaceServerCapabilities,
+};
+
+pub const CODE_ACTION_KIND_SOURCE_FIX_ALL_OXC: CodeActionKind =
+    CodeActionKind::new("source.fixAll.oxc");
+
+pub struct Capabilities {
+    pub code_action_provider: bool,
+}
+
+impl From<ClientCapabilities> for Capabilities {
+    fn from(value: ClientCapabilities) -> Self {
+        // check if the client support some code action literal support
+        let code_action_provider = value.text_document.is_some_and(|capability| {
+            capability.code_action.is_some_and(|code_action| {
+                code_action.code_action_literal_support.is_some_and(|literal_support| {
+                    !literal_support.code_action_kind.value_set.is_empty()
+                })
+            })
+        });
+
+        Self { code_action_provider }
+    }
+}
+
+impl From<Capabilities> for ServerCapabilities {
+    fn from(value: Capabilities) -> Self {
+        Self {
+            text_document_sync: Some(TextDocumentSyncCapability::Kind(TextDocumentSyncKind::FULL)),
+            workspace: Some(WorkspaceServerCapabilities {
+                workspace_folders: Some(WorkspaceFoldersServerCapabilities {
+                    supported: Some(true),
+                    change_notifications: Some(OneOf::Left(true)),
+                }),
+                file_operations: None,
+            }),
+            code_action_provider: if value.code_action_provider {
+                Some(CodeActionProviderCapability::Options(CodeActionOptions {
+                    code_action_kinds: Some(vec![
+                        CodeActionKind::QUICKFIX,
+                        CODE_ACTION_KIND_SOURCE_FIX_ALL_OXC,
+                    ]),
+                    work_done_progress_options: WorkDoneProgressOptions {
+                        work_done_progress: None,
+                    },
+                    resolve_provider: None,
+                }))
+            } else {
+                None
+            },
+            ..ServerCapabilities::default()
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use tower_lsp::lsp_types::{
+        ClientCapabilities, CodeActionClientCapabilities, CodeActionKindLiteralSupport,
+        CodeActionLiteralSupport, TextDocumentClientCapabilities,
+    };
+
+    use super::Capabilities;
+
+    /// this is from build (see help, about):
+    /// Version: 1.95.3 (user setup)
+    /// Commit: f1a4fb101478ce6ec82fe9627c43efbf9e98c813
+    /// Date: 2024-11-13T14:50:04.152Z
+    /// Electron: 32.2.1
+    /// ElectronBuildId: 10427718
+    /// Chromium: 128.0.6613.186
+    /// Node.js: 20.18.0
+    /// V8: 12.8.374.38-electron.0
+    /// OS: Windows_NT x64 10.0.19045
```

Comment:
Does this comment apply to the entire `client_capabilities` variable, or just the code_action_kind?  I would move it to just above the `value_set: vec![` line if that is all it applies to.

I also think the only relevant part is the version and the commit, the rest is probably safe to drop.

Reply:
No only the code_actions check :) 
Moved it. and thanks for your info for intellij